### PR TITLE
Cleanup methods for containers and images

### PR DESCRIPTION
**Pull request by dhodovsk**
This PR implements following functionality:
 - Label containers created with conu - in run_via_binary - should be also invoked in implementation of run_via_api
 - Tag images pulled by conu - comes along with presence check of images before pulling
 - when Prune is run, all containers with conu label and images with conu tag are removed

### CODE REVIEW

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/BoCnd](https://da.gd/BoCnd)__
* Comment: __See details for logs__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/15SB](https://da.gd/15SB)__
* Comment: __See details for logs__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/e05b](https://da.gd/e05b)__
* Comment: __See details for logs__

File: conu/backend/docker/prune.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import logging
+
+from conu.backend.docker.client import get_client
+
+logger = logging.getLogger(__name__)
+CONU_ARTIFACT_TAG='io.conu.test_artifact'
```
Comment (TomasTomecek) [inline conu/backend/docker/prune.py]: damn, that domain is already taken

**Comment by TomasTomecek** [comment]
> WIP: create test for cleanup_images - we need to find/create image that is minimal and not likely to be present at the beginning of the test

Just create one based on an existing image using `run` & `commit`.

Also, what do you think about including the prune API inside the Backend class?

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/wexF](https://da.gd/wexF)__
* Comment: __See details for logs__

File: tests/integration/test_backend.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    backend = DockerBackend(logging_level=logging.DEBUG)
+    client = get_client()
+    container_sum = len(client.containers(all=True))
+    image = DockerImage(FEDORA_MINIMAL_REPOSITORY, tag=FEDORA_MINIMAL_REPOSITORY_TAG)
+    command = DockerRunBuilder(command=["ls"], additional_opts=["-i", "-t"])
+
+    for i in range(3):
+        image.run_via_binary(command)
+
+    assert container_sum+3 == len(client.containers(all=True))
```
Comment (TomasTomecek) [inline tests/integration/test_backend.py]: since this is where CI fails, I suggest applying filter for containers which are younger than 1 minute

**Comment by TomasTomecek** [comment]
Tests are passing. Merging. Well done! (we'll update the label name once we figure it out)

**Pull request merged**